### ISSUE TITLE
`avoid_multiple_declarations_per_line` should skip declarations in for statements

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#2543]
For example:

```dart
for (var i = 0, j = 0; i < foo.length && j < bar.length; ++i, ++j) {
   ...
}
```

shouldn't trigger?

/cc @lukepighetti

**Comment by lukepighetti** [comment]
Just saw this now. Glad #3460 got merged!
